src/pip/_vendor/__init__.py

```diff
@@ -30,24 +30,21 @@ def vendored(modulename):
     vendored_name = "{0}.{1}".format(__name__, modulename)
 
     try:
-        __import__(vendored_name, globals(), locals(), level=0)
+        __import__(modulename, globals(), locals(), level=0)
```

Comment:
By skipping this, are you changing behaviour so that it's no longer possible to debundle only *some* vendored modules? I don't know if this is a valid use case (I don't know anything really about how debundling is implemented or why it's needed/used) but if it is a change in behaviour we should be aware of it.

Reply:
> By skipping this, are you changing behaviour so that it's no longer possible to debundle only some vendored modules?

You're absolutely right.

> I don't know if this is a valid use case

Debian pip maintainer once did that [1]. I believe that's because the `vendored("pep517")` entry was missing [2]. IMO the correct approach should be adding the missing entry like how Arch Linux does [3]. Anyway, mixing system and vendored libraries can lead to unexpected issues [4].

> if it is a change in behaviour we should be aware of it.

I can add a note to the debundling section. Do you prefer that?

[1] https://salsa.debian.org/python-team/modules/python-pip/commit/e7c465a41b2fec16a41b8f07c63e8f6cc26ba817
[2] https://github.com/pypa/pip/pull/6056
[3] https://git.archlinux.org/svntogit/packages.git/commit/trunk?h=packages/python-pip&id=8210fc43e01862e5164cb37780669e456e6a0f52
[4] https://github.com/pypa/pip/issues/5757#issuecomment-424388739

Reply:
I'm perfectly OK with just adding a note. After all, debundling is [only semi-supported](https://pip.pypa.io/en/latest/development/vendoring-policy/#debundling) anyway ;-)

> I believe that's because the vendored("pep517") entry was missing

Hmm, there's nothing in the [vendoring policy](https://pip.pypa.io/en/latest/development/vendoring-policy/#vendoring-policy) that mentions that this is needed. I'd have probably missed it anyway 😉 but it's probably something that should be added, for completeness.

Reply:
I updated the file in this repo. I guess contents on https://pip.pypa.io/en/latest/development/vendoring-policy/#debundling will be synchronized?

> it's probably something that should be added, for completeness.

Let me work on another PR for that!

Reply:
> it's probably something that should be added, for completeness.

Done in #6114.

Reply:
> I guess contents on https://pip.pypa.io/en/latest/development/vendoring-policy/#debundling will be synchronized?

For typing loose ends -- Yes, the sources for that page is the file you've referenced.

It's stored in that directory as a README, to keep the same information more visible on GitHub's UI.